ssz_serialization/merkleization.nim

```diff
@@ -165,42 +177,46 @@ func computeZeroHashes: array[sizeof(Limit) * 8, Digest] =
 
 const zeroHashes* = computeZeroHashes()
 
-template combineChunks(start: int) =
-  for i in start .. merkleizer.topIndex:
+template combineChunks(merkleizer: var SszMerkleizerImpl, start: int) =
+  for i in start..<merkleizer.topIndex:
     trs "CALLING MERGE BRANCHES"
-    doAssert i != merkleizer.topIndex
     if getBitLE(merkleizer.totalChunks, i + 1):
-      merkleizer.combinedChunks[i + 1][1] = mergeBranches(
+      # chunkPtr avoids a temporary in the generated nim code
+      let chunkPtr = addr merkleizer.combinedChunks[i + 1][1]
```

Comment:
Ideally, there will be a link to a Nim issue that's being worked-around here.

Reply:
there are many lack-of-RVO bugs in Nim - it's a long journey to isolate each and every one - instead, https://github.com/nim-lang/Nim/issues/19357 collects it as a high-level issue because this needs a complete review in the compiler rather than a whack-a-mole